google-cloud-logging/src/main/java/com/google/cloud/logging/MonitoredResourceUtil.java

```diff
@@ -193,18 +195,47 @@ private static String getAppEngineInstanceName() {
   }
 
   private static List<LoggingEnhancer> getEnhancers(Resource resourceType) {
-    List<LoggingEnhancer> enhancers;
+    List<LoggingEnhancer> enhancers = new ArrayList<>();
     switch (resourceType) {
       // Trace logging enhancer is supported on GAE Flex and Standard.
-      case GaeAppStandard:
       case GaeAppFlex:
-        enhancers = new ArrayList<>();
-        enhancers.add(new TraceLoggingEnhancer());
+        enhancers.add(new LabelLoggingEnhancer(
+            APPENGINE_LABEL_PREFIX, Collections.singletonList(Label.InstanceName)));
+        enhancers.add(new TraceLoggingEnhancer(APPENGINE_LABEL_PREFIX));
+        break;
+      case GaeAppStandard:
+        enhancers.add(new TraceLoggingEnhancer(APPENGINE_LABEL_PREFIX));
         break;
       default:
         enhancers = Collections.emptyList();
         break;
     }
     return enhancers;
   }
+
+  /**
+   *  Adds additional resource-based labels to log entries.
+   *  Labels that can be provided with {@link MonitoredResource.Builder#addLabel(String, String)}
+   *  are restricted to a supported set per resource.
+   *  @see <a href="https://cloud.google.com/logging/docs/api/v2/resource-list">Logging Labels</a>
+   */
+  private static class LabelLoggingEnhancer implements LoggingEnhancer {
+    String prefix;
+    List<Label> labels;
+
+    LabelLoggingEnhancer(String prefix, List<Label> labels) {
```

Comment:
Storing a reference to an outside collection as a field in a class is usually undesirable (as the outside reference can be changed independently at any time, which can lead to unpredictable results, for example ConcurrentModificationException during iteration over the list).

Here, it looks like this enhancer needs a prefixed label name and a corresponding constant value (retrieved by MonitoredResourceUtil.getValue()). Both of these values are constants. I don't know how important performance is for this particular code, switching to a map (with prefix + label.getKey() as a key and MonitoredResourceUtil.getValue(label) as value), can solve the two potential issues, not affecting readability of the code:

possibility of unexpected modification of the labels list;
slightly worse performance, caused by prefix + label.getKey() addition (note, two constants are getting added over and over again on each execution of the enhanceLogEntry method).